### PR TITLE
Test Pulp's validation of rsync dist. config opts.

### PR DESCRIPTION
**Pull request by Ichimonji10**
Add a new test case, `VerifyOptionsTestCase`, that test Pulp's
validation of the RPM rsync distributor configuration options. This test
case targets the following issues:

* Pulp issue 2191: The RPM rsync distributor's "predistributor_id"
  config option isn't validated
* Pulp issue 2192: The RPM rsync distributor's "root" config option
  isn't validated

See:

* https://pulp.plan.io/issues/2191
* https://pulp.plan.io/issues/2192

### CODE REVIEW

File: pulp_smash/tests/rpm/api_v2/test_rsync_distributor.py [hunk 1]
```diff
@@ -269,3 +270,94 @@ def test_all(self):
...
+                'predistributor_id': utils.uuid4(),
+                'remote': self.remote,
+            }
+        })
+        try:
+            with self.assertRaises(HTTPError, msg=body):
+                repo_href = api_client.post(REPOSITORY_PATH, body)['_href']
+        except AssertionError as err:
+            self.addCleanup(api_client.delete, repo_href)
+            raise err
```
Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_rsync_distributor.py]: This try/except block really does work. You can verify by dropping the `selectors` block at the top of this method.

**Comment by Ichimonji10** [comment]
Pulp 2.10 results:

```
(pulp-smash3) [ichimonji10@beech:pulp-smash]$ python -m unittest2 pulp_smash.tests.rpm.api_v2.test_rsync_distributor.VerifyOptionsTestCase
s..s.
----------------------------------------------------------------------
Ran 5 tests in 9.582s

OK (skipped=2)
```

Pulp 2.9 results:

```
(pulp-smash2) [ichimonji10@beech:pulp-smash]$ python -m unittest2 pulp_smash.tests.rpm.api_v2.test_rsync_d
istributor
s
----------------------------------------------------------------------
Ran 0 tests in 1.637s

OK (skipped=1)
```

**Comment by Ichimonji10** [comment]
If you're a reviewer, this might help: https://github.com/pulp/pulp_rpm/blob/master/docs/tech-reference/rsync-distributor.rst

File: pulp_smash/tests/rpm/api_v2/test_rsync_distributor.py [hunk 2]
```diff
@@ -269,3 +270,94 @@ def test_all(self):
...
+    """Test Pulp's verification of RPM rsync distributor configuration options.
+
+    Do the following:
+
+    * Repeatedly attempt to create an RPM repository with an importer and pair
+      of distributors. Each time, pass an invalid option, or omit a required
+      option.
+    * Create an RPM repository with an importer and pair of distributors. Pass
+      valid configuration options. This demonstrates that creation failures are
+      do to Pulp's validation logic, not some other factor.
```
Comment (elyezer) [inline pulp_smash/tests/rpm/api_v2/test_rsync_distributor.py]: s/do/due?

**Pull request merged**